lib/presentation/view/top_rated.dart

```diff
@@ -47,45 +51,7 @@ class _TopRatedState extends State<TopRated> {
                 valueColor: AlwaysStoppedAnimation<Color>(Palette.primaryLight),
               );
             case Status.success:
-              return ListView.builder(
-                itemCount: snapshot.data!.movies!.length,
-                itemBuilder: (
-                  BuildContext context,
-                  int index,
-                ) {
-                  return Padding(
-                    padding:
-                        const EdgeInsets.all(Dimensions.movieContainerPadding),
-                    child: Container(
-                      color: Palette.containerColor,
-                      child: Padding(
-                        padding: const EdgeInsets.all(
-                          Dimensions.innerMovieContainerPadding,
-                        ),
-                        child: Row(
-                          children: [
-                            showImgIfExists(
-                              snapshot.data!.movies![index].posterPath,
-                              Dimensions.moviePosterSize,
-                            ),
-                            Expanded(
-                              child: AutoSizeText(
-                                snapshot.data!.movies![index].title,
-                                style: const TextStyle(
-                                  color: Palette.lightFontColor,
-                                  fontSize: Dimensions.movieTitleFontSize,
-                                ),
-                                textAlign: TextAlign.center,
-                                maxLines: Dimensions.movieTitleMaxLines,
-                              ),
-                            ),
-                          ],
-                        ),
-                      ),
-                    ),
-                  );
-                },
-              );
+              return buildMovieList(snapshot.data!);
```

Comment:
Instead to send data through the method and you are in the Success event, you can send only the list

Reply:
buildMovieList now takes a list of MovieModel instead of a MovieEvent